code/game/atoms.dm

```diff
@@ -361,40 +361,128 @@
 		A.fingerprintshidden |= fingerprintshidden.Copy()    //admin
 	A.fingerprintslast = fingerprintslast
 
+var/list/blood_splatter_icons = list()
 
-//returns 1 if made bloody, returns 0 otherwise
-/atom/proc/add_blood(mob/living/carbon/human/M as mob)
+/atom/proc/blood_splatter_index()
+	return "\ref[initial(icon)]-[initial(icon_state)]"
 
-	if(!blood_DNA || !istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
-		blood_DNA = list()
+//returns the mob's dna info as a list, to be inserted in an object's blood_DNA list
+/mob/living/proc/get_blood_dna_list()
+	if(get_blood_id() != "blood")
+		return
+	return list("ANIMAL DNA" = "Y-")
 
-	blood_color = "#A10808"
-	if(istype(M))
-		if(M.species.flags & NO_BLOOD)
-			return 0
-		M.check_dna()
-		blood_color = M.species.blood_color
+/mob/living/carbon/get_blood_dna_list()
+	if(get_blood_id() != "blood")
+		return
+	var/list/blood_dna = list()
+	if(dna)
+		var/mob/living/carbon/human/H = src
+		blood_dna[dna.unique_enzymes] = H.b_type
+		blood_dna["blood_color"] = H.species.blood_color
+	else
+		blood_dna["UNKNOWN DNA"] = "X*"
+	return blood_dna
 
-	. = 1
-	return 1
+/mob/living/carbon/alien/get_blood_dna_list()
+	return list("UNKNOWN DNA" = "X*")
 
-/atom/proc/add_blood_list(mob/living/carbon/M)
+//to add a mob's dna info into an object's blood_DNA list.
+/atom/proc/transfer_mob_blood_dna(mob/living/L)
 	// Returns 0 if we have that blood already
-	if(!istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
+	var/new_blood_dna = L.get_blood_dna_list()
+	if(!new_blood_dna)
+		return 0
+	if(!blood_DNA)	//if our list of DNA doesn't exist yet, initialise it.
 		blood_DNA = list()
-	//if this blood isn't already in the list, add it
-	if(blood_DNA[M.dna.unique_enzymes])
-		return 0 //already bloodied with this blood. Cannot add more.
-	var/blood_type = "X*"
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		blood_type = H.b_type
-	blood_DNA[M.dna.unique_enzymes] = blood_type
+	var/old_length = blood_DNA.len
+	blood_DNA |= new_blood_dna
+	if(blood_DNA.len == old_length)
+		return 0
 	return 1
 
-// Only adds blood on the floor -- Skie
-/atom/proc/add_blood_floor(mob/living/carbon/M)
-	return //why the fuck this is at an atom level but only works on simulated turfs I don't know
+/obj/effect/decal/cleanable/blood/splatter/transfer_mob_blood_dna(mob/living/L)
+	..(L)
+	var/list/b_data = L.get_blood_data(L.get_blood_id())
+	basecolor = b_data["blood_color"]

```

Comment:
wel its adding the basecolor to the decal and update_icon sets the color..so yeah i could try adding an update_icon proc to it. but it does need the basecolor set.